auction-server/src/kernel/entities.rs

```diff
@@ -39,3 +52,41 @@ impl<'de> Deserialize<'de> for PermissionKeySvm {
         Ok(PermissionKeySvm(bytes))
     }
 }
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum ChainType {
+    Evm,
+    Svm,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Evm;
+
+impl Evm {
+    pub fn decode_logs_for_receipt(receipt: &TransactionReceipt) -> Vec<MulticallIssuedFilter> {
```

Comment:
let's move this to the same file as the caller. No need to put it inside some impl. Feels very weird here